vm/src/scope.rs

```diff
@@ -97,19 +47,22 @@ impl Scope {
     }
 
     pub fn get_locals(&self) -> PyDictRef {
-        match self.locals.iter().next() {
+        match self.locals.first() {
             Some(dict) => dict.clone(),
             None => self.globals.clone(),
         }
     }
 
     pub fn get_only_locals(&self) -> Option<PyDictRef> {
-        self.locals.iter().next().cloned()
+        self.locals.first().cloned()
     }
 
     pub fn new_child_scope_with_locals(&self, locals: PyDictRef) -> Scope {
+        let mut new_locals = Vec::with_capacity(self.locals.len() + 1);
+        new_locals.push(locals);
+        new_locals.append(&mut self.locals.clone());
```

Comment:
```suggestion
        new_locals.extend_from_slice(&self.locals);
```

Reply:
I changed for `extend_from_slice` but I don't think I need to call `shrink_to_fit` as I create the vector with the exact needed capacity.